src/components/ServiceDetails/index.js

```diff
@@ -36,7 +37,14 @@ class ServiceDetails extends Component {
 
   populateServiceData = () => {
     const { service_row_id } = this.props.match.params;
-    const service = this.props.services.filter(el => el.service_row_id === Number(service_row_id))[0];
+    let service = this.props.services.filter(el => el.service_row_id === Number(service_row_id))[0];
+
+    if (typeof service !== "undefined") {
+      const pricing = service["pricing"];
+      let pricingJSON = typeof pricing === "undefined" || pricing === null ? JSON.stringify(service) : pricing;
```

Comment:
What we could do is parse the service object right when we get the response which has all the information needed rather than doing price parsing here. And from there on we can pass the parsed service object with all the needed information through the app.

Reply:
Done. Handling it at the source